c10/core/DeviceCapability.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <c10/macros/Export.h>
+#include <c10/core/ScalarType.h>
+#include <cstdint>
+#include <vector>
+
+namespace c10 {
+
+  constexpr size_t NUMBER_OF_DEVICE_CAPABILITIES = NumScalarTypes;
+
+// Generate bitfields for each scalar type
+#define DEFINE_SCALAR_TYPE(_1, n) unsigned int has_##n : 1;
+
+struct C10_API DeviceCapability {
```

Comment:
We need to add some description for `DeviceCCapability` to show the its purpose, contract and how to add new capability for it.